typing/ctype.mli

```diff
@@ -220,7 +220,7 @@ val instance_poly_fixed:
         (* Take an instance of a type scheme containing free univars for
            checking that an expression matches this scheme. *)
 
-val polyfy: Env.t -> type_expr -> type_expr list -> type_expr * bool
+val polyfy: Env.t -> type_expr -> type_expr list -> type_expr * type_expr list
```

Comment:
This type feels a bit strange, I guess it's something like `normal_result * error`.
Are the two mutually exclusive? Or does the first member of the tuple only make sense when the second is the empty list?

Reply:
In presence of errors the left `type_expr` contains the "partially quantified" version of the the type expression, 
```ocaml
Error: This definition has type 'd 'e. 'd -> 'e -> 'd
       which is less general than 'a 'b 'c. 'a -> 'b -> 'c
```
thus the type is more `quantified_type_expr * variable_that_could_not_lifted list` and an alternative type could be `(type_expr, type_expr * type_expr nonempty_list) result`.